src/kaleidoscope/hardware/ez/ErgoDox.h

```diff
@@ -26,6 +26,7 @@
 #pragma once
 
 #ifdef ARDUINO_AVR_ERGODOX
+#ifndef KALEIDOSCOPE_VIRTUAL_BUILD
```

Comment:
The device description class as well as all header includes that the device description's definition depends upon and also the `PER_KEY_DATA_STACKED` must live outside of the `#ifndef KALEIDOSCOPE_VIRTUAL_BUILD` clause to be available to the virtual build when `KALEIDOSCOPE_VIRTUAL_BUILD` is defined.

Also, all hardware headers must define a global typedef e.g. `DeviceDescription` (possibly in kaleidoscope namespace) that can be picked up by the virtual build. Sorry that I forgot to mention the latter earlier.

Reply:
After a bit of pondering, my current plan is to remove the virtual build ifdefs for the duration of this PR, and add them in another, followup PR soon after. The main reason is that there's already plenty of things to review here, and even I have trouble keeping the changes all in mind. It'll hopefully be easier and clearer to fix up virtual builds as a next step.